test/python/pulse/test_schedule.py

```diff
@@ -518,31 +564,77 @@ def test_filter_multiple(self):
         sched = sched.insert(60, acquire(device.acquires, device.memoryslots))
         sched = sched.insert(90, lp0(device.drives[0]))
 
-        filtered = sched.filter(channels={0}, instruction_types=[PulseInstruction],
-                                time_ranges=[(25, 100)])
+        # split instructions with filters on channel 0, of type PulseInstruction,
+        # occuring in the time interval (25, 100)
+        filtered, excluded = self.filter_and_test_consistency(sched,
+                                                              channels={device.drives[0]},
+                                                              instruction_types=[PulseInstruction],
+                                                              time_ranges=[(25, 100)])
         for time, inst in filtered.instructions:
             self.assertIsInstance(inst, PulseInstruction)
-            self.assertTrue(any([chan.index == 0 for chan in inst.channels]))
+            self.assertTrue(all([chan.index == 0 for chan in inst.channels]))
             self.assertTrue(25 <= time <= 100)
-
-        filtered_b = sched.filter(instruction_types=[PulseInstruction, FrameChangeInstruction],
-                                  time_ranges=[(25, 100), (0, 30)])
-        for time, inst in filtered_b.instructions:
+        self.assertEqual(len(excluded.instructions), 4)
+        self.assertTrue(excluded.instructions[0][1].channels[0] == DriveChannel(0))
+        self.assertTrue(excluded.instructions[2][0] == 30)
+
+        # split based on PulseInstructions in the specified intervals
+        filtered, excluded = self.filter_and_test_consistency(sched,
+                                                              instruction_types=[PulseInstruction],
+                                                              time_ranges=[(25, 100), (0, 11)])
+        self.assertTrue(len(excluded.instructions), 3)
+        for time, inst in filtered.instructions:
             self.assertIsInstance(inst, (FrameChangeInstruction, PulseInstruction))
-        self.assertTrue(len(filtered_b.instructions), 4)
+        self.assertTrue(len(filtered.instructions), 4)
+        # make sure the PulseInstruction not in the intervals is maintained
+        self.assertIsInstance(excluded.instructions[0][1], PulseInstruction)
 
-    def test_filter(self):
-        """Test _filter method."""
+    def test_custom_filters(self):
+        """Test custom filters."""
         device = self.two_qubit_device
         lp0 = self.linear(duration=3, slope=0.2, intercept=0.1)
         sched = Schedule(name='fake_experiment')
         sched = sched.insert(0, lp0(device.drives[0]))
         sched = sched.insert(10, lp0(device.drives[1]))
         sched = sched.insert(30, FrameChange(phase=-1.57)(device.drives[0]))
-        for i in sched._filter([lambda x: True]).instructions:
+
+        filtered, excluded = self.filter_and_test_consistency(sched, [lambda x: True])
```

Comment:
Ah, so, this should not be passed as a list. The `*` before `filter_funcs` splices the args for you. So, you'd write `self.filter...tency(sched, lambda x: True)` and then the variable `filter_funcs` will be assigned to `[lambda x: True]`.

Then you can remove those extra processing lines you had above.